src/test/java/teammates/test/pageobjects/StudentProfilePicturePage.java

```diff
@@ -15,7 +15,16 @@ public StudentProfilePicturePage(Browser browser) {
 
     @Override
     protected boolean containsExpectedPageContents() {
-        return true;
+        String pageSource = getPageSource();
+
+        // First test is for the actual pages, the tests for after the first || is meant for tests that
+        // results in error pages or entity not found pages, note that some of them do not have tags,
+        // have no closing tags and those are intentional
+        return pageSource.contains("<h1>Student Profile</h1>")
+               || pageSource.contains("<title>studentProfilePic")
+               || pageSource.contains("The page you are looking for is not there.")
+               || pageSource.contains("You are not authorized to view this page.")
+               || pageSource.contains("TEAMMATES could not locate what you were trying to access.");
```

Comment:
Yeah, I'm not exactly sure either, I believe this is an improvement over "return true;" but we may need more eyes on this, perhaps it can help to generate better ideas?